src/propmcwf.jl

```diff
@@ -0,0 +1,92 @@
+immutable QuMCWF <: QuPropagatorMethod
+end
+
+type QuMCWFEnsemble{QA<:QuBase.AbstractQuArray}
+    ntraj::Int
+    rho::QA
+    decomp
+    QuMCWFEnsemble(ntraj, rho, decomp) = new(ntraj, rho, decomp)
+end
+
+QuMCWFEnsemble{QM<:QuBase.AbstractQuMatrix}(ntraj::Int, rho::QM) = QuMCWFEnsemble{QM}(ntraj, rho, eigfact(full(coeffs(rho))))
+QuMCWFEnsemble{QV<:QuBase.AbstractQuVector}(ntraj::Int, psi::QV) = QuMCWFEnsemble{QV}(ntraj, psi, nothing)
+QuMCWFEnsemble{QM<:QuBase.AbstractQuMatrix}(rho::QM) = QuMCWFEnsemble{QM}(500, rho, eigfact(full(coeffs(rho))))
+QuMCWFEnsemble{QV<:QuBase.AbstractQuVector}(psi::QV) = QuMCWFEnsemble{QV}(500, psi, nothing)
+
+Base.start(mcwfensemble::QuMCWFEnsemble) = 1
+
+function Base.next(mcwfensemble::QuMCWFEnsemble, i::Int)
+    state = draw(mcwfensemble)
+    return (state, i+1)
+end
+
+function Base.done(mcwfensemble::QuMCWFEnsemble, i::Int)
+    i > mcwfensemble.ntraj
+end
+
+function draw{QM<:QuBase.AbstractQuMatrix}(mcwfensemble::QuMCWFEnsemble{QM})
+    r = rand() # draw random number from [0,1)
+    pacc = 0.
+    for i=1:length(mcwfensemble.decomp[:values])
+        pacc = pacc + mcwfensemble.decomp[:values][i]
+        if pacc >= r
+            return QuArray(complex(mcwfensemble.decomp[:vectors][:,i]))
+        end
+    end
+end
+
+function draw{QV<:QuBase.AbstractQuVector}(mcwfensemble::QuMCWFEnsemble{QV})
+    return copy(mcwfensemble.rho)
+end
+
+function eff_hamiltonian(lme::QuLindbladMasterEq)
+    heff = lme.hamiltonian
+    for c_ops in lme.collapse_ops
+       heff = heff - im*0.5*c_ops'*c_ops
+    end
+    return heff
+end
+
+function propagate(prob::QuMCWF, eq::QuLindbladMasterEq, t, current_t, psi)
+    jtol = 1.e-6  # jump tolerance
+    # get information of QME
+    heff = eff_hamiltonian(eq)
+    c_ops = eq.collapse_ops
+    eps = rand() # draw random number from [0,1)
+    dt = t - current_t
+    accdt = 0.
+    tj = 0.
+    while accdt < dt
+        # propagate one time-step
+        psi1 = expm(-im*heff*dt)*psi
```

Comment:
I think we have to use some other method here. Basically, we need an option for `QuMCWF`, which allows us to specify the propagator we want to use for this. Maybe we can just use the propagator type as an option and then call `propagate` directly (instead of using the iterator interface)?